src/Domain/FunctionsOfTime/PiecewisePolynomial.cpp

```diff
@@ -282,17 +215,28 @@ template <size_t MaxDeriv>
 std::ostream& operator<<(
     std::ostream& os,
     const PiecewisePolynomial<MaxDeriv>& piecewise_polynomial) {
-  const size_t writable_size =
-      piecewise_polynomial.deriv_info_size_.load(std::memory_order_acquire);
-  auto it = piecewise_polynomial.deriv_info_at_update_times_.begin();
-  // Can't use .end() because of thread-safety and don't want to do expensive
-  // std::next() call, so we just loop over allowed size while incrementing an
-  // iterator
-  for (size_t i = 0; i < writable_size; i++, it++) {
-    os << *it;
-    // writable_size guaranteed to be >= 1. No need to worry about negative
-    // size_t
```

Comment:
Can you keep this comment in?

Reply:
`writable_size` doesn't exist anymore.  I also don't really understand what the comment was trying to tell me.

Reply:
You have
```cpp
    if (entry != iters.size() - 1) {
      os << "\n";
    }
```
and if `iters.size() == 0` this is bad. But we know that can't happen for other reasons. So replace `writable_size` with `iters.size()` in the comment

Reply:
Why would that be bad?  If this was a `<` test or similar it could be confusing, but `!=` does exactly what you'd expect.

Somewhat unrelated: Now that I look at it more, I should probably comment about why we know `std::next([...].begin())` is OK elsewhere in the function.